### ISSUE TITLE
Conditional on include task fails

### ISSUE THREAD

**Issue by andreea-vld** [ansible/ansible#17702]
##### ISSUE TYPE

- Bug Report
##### COMPONENT NAME

conditional include
##### ANSIBLE VERSION

```
ansible 2.1.1.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides

```
##### CONFIGURATION

n/a
##### OS / ENVIRONMENT

Ubuntu 14.04
##### SUMMARY

I have an include task conditioned by a when clause which resolves to True, but the playbook is not included. If I add the SAME condition in a variable and use that variable in _when_ , it is evaluated to true and the playbook is included.
##### STEPS TO REPRODUCE

Run site.yml with --extra-vars '{"scale"}': <br>
**ansible-playbook --extra-vars '{"scale"}' site.yml** <br>
site.yml that fails (skips acq.yml):

```
- name: Top-level playbook
  hosts: localhost
  pre_tasks:
    - name: Check dev type
      set_fact: DEV=true
      when: scale is not defined

    - name: Check test type
      set_fact: TEST=true
      when: scale is defined

- include: acq.yml
  when: ((TEST is defined) or (DEV is defined and (USE_ACQ is defined or USE_3DS is defined)))
```

Run site2.yml with --extra-vars '{"scale"}': <br>
**ansible-playbook --extra-vars '{"scale"}' site2.yml** <br>
site2.yml that passes (executes acq.yml): <br>

```
- name: Top-level playbook
  hosts: localhost
  pre_tasks:
    - name: Check dev type
      set_fact: DEV=true
      when: scale is not defined

    - name: Check test type
      set_fact: TEST=true
      when: scale is defined

    - name: Set var
      set_fact: var={{((TEST is defined) or (DEV is defined and (USE_ACQ is defined or USE_3DS is defined)))}}

- include: acq.yml
  when: var
```

##### EXPECTED RESULTS

In both cases, run acq.yml
##### ACTUAL RESULTS

For site.yml, acq.yml is skipped <br>
For site2.yml, acq.yml is executed

```
skipping playbook include 'acq.yml' due to conditional test failure
```

**Comment by philipgough** [comment]
+1 Seeing this also now in 2.2.0.0

**Comment by genei09** [comment]
I've had a similar experience with 2.2.0.0

the following always executes regardless of the value of build_type
```
- name: "Finalize setup"
  when: build_type == 'dev'
  include: ../tasks/app_setup.yml
```

as a workaround the following works as expected
```
- name: "Finalize setup"
  when: build_type == 'dev'
  include: ../tasks/{{ notapp | default('app') }}_setup.yml
```

**Comment by bcoca** [comment]
try
```
- include: acq.yml
  when: var
  static: no
```

includes by default are not affected by `when`, only when they are dynamic, otherwise the condition is  just inherited by the included tasks.

**Comment by bcoca** [comment]
Possible Misunderstanding
=========================

Hi!

Thanks very much for your submission to Ansible.  It sincerely means a lot to us.

We believe the ticket you have filed is being somewhat misunderstood, as one thing works a little differently than stated.

In the future, this might be a topic more well suited for the user list, which you can also post here if you'd like some more help with the above.
   
   * https://groups.google.com/forum/#!forum/ansible-project

Thank you once again for this and your interest in Ansible!

**Comment by jimi-c** [comment]
This has nothing to do with static includes, however this should be fixed in both devel and stable-2.2 (and the 2.2.1 RC2).
<pre>
[jamesc@jimi 17702]$ ansible-playbook --version
ansible-playbook 2.2.1.0 (stable-2.2 d8449b3013) last updated 2016/12/16 11:20:09 (GMT -500)
  lib/ansible/modules/core: (detached HEAD 62dbe0debb) last updated 2016/12/19 15:20:56 (GMT -500)
  lib/ansible/modules/extras: (detached HEAD 17eebf2b4f) last updated 2016/12/19 15:20:56 (GMT -500)
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/usr/share/ansible']

[jamesc@jimi 17702]$ ansible-playbook -vv --extra-vars '{"scale"}' test.yml 
Using /etc/ansible/ansible.cfg as config file

PLAYBOOK: test.yml *************************************************************
2 plays in test.yml

PLAY [Top-level playbook] ******************************************************

TASK [setup] *******************************************************************
ok: [localhost]

TASK [Check dev type] **********************************************************
task path: /home/jamesc/testing/17702/test.yml:4
skipping: [localhost] => {"changed": false, "skip_reason": "Conditional check failed", "skipped": true}

TASK [Check test type] *********************************************************
task path: /home/jamesc/testing/17702/test.yml:8
ok: [localhost] => {"ansible_facts": {"TEST": true}, "changed": false}

PLAY [localhost] ***************************************************************

TASK [debug] *******************************************************************
task path: /home/jamesc/testing/17702/acq.yml:4
ok: [localhost] => {
    "msg": "here's the included playbook"
}

PLAY RECAP *********************************************************************
localhost                  : ok=3    changed=0    unreachable=0    failed=0   
</pre>
Thanks!
